ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -173,12 +173,8 @@ impl<Register: Send + Sync + 'static> TraceBuilder<Register> {
                 aot_ir::Inst::Promote { tyidx, .. }
                 | aot_ir::Inst::IdempotentPromote { tyidx, .. } => {
                     // Consume the correct number of bytes from the promoted values array.
-                    let width_bits = match self.aot_mod.type_(*tyidx) {
-                        aot_ir::Ty::Integer(x) => x.bitw(),
-                        _ => unreachable!(),
```

Comment:
Here we would encounter a `Ty::Pointer` and panic.

Reply:
Got it. It was hidden by the (not necessarily bad, but not exactly key) refactoring.